modules/detection/VADAudioAnalyser.js

```diff
@@ -0,0 +1,219 @@
+import { EventEmitter } from 'events';
+import { getLogger } from 'jitsi-meet-logger';
+
+import * as JitsiConferenceEvents from '../../JitsiConferenceEvents';
+
+import { VAD_SCORE_PUBLISHED, DETECTOR_STATE_CHANGE } from './DetectionEvents';
+import TrackVADEmitter from './TrackVADEmitter';
+
+const logger = getLogger(__filename);
+
+/**
+ * Sample rate of TrackVADEmitter, it defines how many audio samples are processed at a time.
+ * @type {number}
+ */
+const VAD_EMITTER_SAMPLE_RATE = 4096;
+
+/**
+ * Connects a TrackVADEmitter to the target conference local audio track and manages various services that use
+ * the data to produce audio analytics (VADTalkMutedDetection and VADNoiseDetection).
+ */
+export default class VADAudioAnalyser extends EventEmitter {
```

Comment:
Hmm... well the thing is that `VADAudioAnalyser` no longer has anything to do with `VADNoiseDetection` and `VADTalkMutedDetection`, that was actually the whole point i tried to apply the COI principle and decouple as much as I could. So you could use the detector outside the scope of the VADAudioAnalyser if you wish, and replace them with ease inside `VADAudioAnalyser` as long as they stick to the interface (duck typed as it may be ) which in term allows for easier mocking and testing if need be, component based code reuse is much more easier to test IMO. 
On the other hand `VADTalkMutedDetection` and `VADAudioAnalyser` kind of make sense to be inherited from a base class to establish an interface, but from what i saw up until this point this is just left to duck typing in javascript. And i didn't feel there was enough reusable stuff there to be moved to a common parent as they behave differently and wouldn't really abide by the Liskov principle, however that might be a bit subjective from my part so, if this was what you meant i can move a couple of things around.